src/redisearch_api.c

```diff
@@ -631,3 +638,91 @@ void RediSearch_SetCriteriaTesterThreshold(size_t num) {
 int RediSearch_StopwordsList_Contains(RSIndex* idx, const char *term, size_t len) {
   return StopWordList_Contains(idx->stopwords, term, len);
 }
+
+void RediSearch_FieldInfo(struct RSIdxField *infoField, FieldSpec *specField) {
+  infoField->name = rm_strdup(specField->name);
+  infoField->path = rm_strdup(specField->path);
+  if (specField->types == INDEXFLD_T_FULLTEXT) {
+    infoField->types.text = true;
+    infoField->textWeight = specField->ftWeight;
+  }
+  if (specField->types == INDEXFLD_T_NUMERIC) {
+    infoField->types.numeric = true;
+  }
+  if (specField->types == INDEXFLD_T_TAG) {
+    infoField->types.tag = true;
+    infoField->tagSeperator = specField->tagSep;
+    infoField->tagCaseSensitive = specField->tagFlags & TagField_CaseSensitive ? 1 : 0;
+  }
+  if (specField->types == INDEXFLD_T_GEO) {
```

Comment:
shouldn't we be checking those doggies bitwise?

Reply:
I agree, flags seems more suitable here.